code/builders/builder-vite/src/build.ts

```diff
@@ -28,5 +35,18 @@ export async function build(options: Options) {
   }).build;
 
   const finalConfig = await presets.apply('viteFinal', config, options);
+
+  const turbosnapPlugin = findPlugin(finalConfig, 'rollup-plugin-turbosnap');
+  if (turbosnapPlugin) {
```

Comment:
I'm not sure if we should maybe not do this is `--webpack-stats-json` isn't passed. That would be inconsistent, and maybe confusing. 

But it changes the behaviour if we drop the plugin if you don't pass the flag (that still wrote the stats file previously).